sourcegraph/sourcegraph
# [Request access] Move `experimentalFeatures.accessRequest` to `auth.accessRequest` config object field

Part of https://github.com/sourcegraph/sourcegraph/issues/48254.

This PR:
- Updates access request feature to use configuration instead of experimental feature
- It also makes configuration an object as `{ disabled?: boolean }`, as later there will be additional ` { allowedEmailDomains: string[] }` field

See also [slack conversation](https://sourcegraph.slack.com/archives/C04LVGAL750/p1679574031699799?thread_ts=1679524862.943469&cid=C04LVGAL750)

## Test plan
- Tests updated
- `sg start`
- Disable built-in signup
- Open https://sourcegraph.test:3443/request-access in incognito mode, the page should be rendered and not redirected anywhere
- Set `auth.accessRequest: {"disabled": true}` in site admin
- Open https://sourcegraph.test:3443/request-access in incognito and it should redirect to sign-in page
